utils/cloudevent.go

```diff
@@ -85,8 +86,15 @@ func DoCloudEventOnce(handler Handler, ctx context.Context, in io.Reader, out io
 		ctx, cancel := CtxWithDeadline(ctx, ceIn.Extensions.Deadline)
 		defer cancel()
 
-		handler.Serve(ctx, strings.NewReader(ceIn.Data.(string)), &resp)
+		if ceIn.ContentType == "application/json" {
+			data := ceIn.Data.(map[string]interface{})
+			err = json.NewEncoder(inBody).Encode(data)
+			handler.Serve(ctx, inBody, &resp)
+		} else {
+			handler.Serve(ctx, strings.NewReader(ceIn.Data.(string)), &resp)
+		}
```

Comment:
ruby fdk appears to expose event id and content type from the cloud event, don't think we're doing that here (on the context?)